src/main/java/org/apache/commons/collections4/bloomfilter/SimpleBloomFilter.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.collections4.bloomfilter;
+
+import java.util.function.Function;
+
+import org.apache.commons.collections4.bloomfilter.hasher.DynamicHasher;
+import org.apache.commons.collections4.bloomfilter.hasher.Hasher;
+import org.apache.commons.collections4.bloomfilter.hasher.Shape;
+import org.apache.commons.collections4.bloomfilter.hasher.function.Murmur128x64Cyclic;
+
+/**
+ * A bloom filter that uses a BitSetBloomFilter to create a Bloom filter that
+ * can merge instances of a specific class.
+ *
+ * @param <T> The Class to merge.
+ * @since 4.5
+ */
+public class SimpleBloomFilter<T> extends BitSetBloomFilter implements BloomFilter {
+
+    /**
+     * The function that converts the instance of T to the SimpleBuilder.
+     * <p>
+     * If the object T is to be considered as a single item in the filter then
+     * function must create the {@code SimpleBuilder} and only call a single {@code with()}
+     * method.</p>
+     * <p>
+     * If the object T is to be considered as several items then the function must
+     * create the {@code SimpleBuilder} and call the {@code with()} method once for each item.</p>
+     */
+    private Function<T, SimpleBuilder> func;
+
+    /**
+     * Constructs a SimpleBloomFilter from the shape and function.  This constructor
+     * creates an empty Bloom filter.
+     * @param hasher the Hasher to use.
+     * @param shape the Shape of the Bloom filter.
+     * @param func a Function to convert T to a SimpleBuilder.
+     * @see #func
+     */
+    public SimpleBloomFilter(Hasher hasher, Shape shape, Function<T, SimpleBuilder> func) {
+        super(hasher, shape);
+        this.func = func;
+    }
+
+    /**
+     * Constructs a SimpleBloomFilter from the shape and function.  This constructor
+     * creates an empty Bloom filter.
+     * @param shape the Shape of the Bloom filter.
+     * @param func a Function to convert T to a SimpleBuilder.
+     * @see #func
+     */
+    public SimpleBloomFilter(Shape shape, Function<T, SimpleBuilder> func) {
+        super(shape);
+        this.func = func;
+    }
+
+    /**
+     * Constructs a SimpleBloomFilter from the shape, function and a data object.
+     * This constructor creates an Bloom filter populated with the data from the
+     * {@code data} parameter.
+     * @param shape the Shape of the Bloom filter.
+     * @param func a Function to convert T to a SimpleBuilder.
+     * @param data the data object to populate the filter with.
+     * @see #func
+     */
+    public SimpleBloomFilter(Shape shape, Function<T, SimpleBuilder> func, T data) {
+        this(shape, func);
+        this.merge( data );
```

Comment:
No extra spaces please: `this.merge( data );` -> `this.merge(data);`

Reply:
As before, creating a filter and adding data are two separate functions. E.g. there is no constructor for an ArrayList that accepts a single item as well as the capacity. 

This concept is better removed or changed to create a filter with another Bloom filter (i.e. a collection). Here this highlights the dual functionality of the Shape which specifies not only the number of hash functions per item but also contains the number of expected items which is not strictly necessary for the filter to function. Creating from another Bloom filter can copy the other filter shape (indeed it mush match hash function identity and number of hash functions) but what would be the expected number of items?

Reply:
I am more asking to remove clutter from the API. It does not seem essential to me to have this constructor that saves 1 line of code.

When using a Bloom filter it would be expected to add hundreds or thousands of items. This will be done in a loop or over time by a method. Adding a single item at the creation point is a drop in the ocean against the final result of the saturated filter.

This is where my example of the ArrayList comes in (or any collection for that matter). There are no collections in the JDK that are created with an item except the singleton specialisations. Thus the language do not see this as a common case to satisfy. I'd suggest dropping the constructor that takes an item.

I would not add a constructor that accepts a Bloom filter since that also only saves 1 line of code over just calling merge on the next line.